### PR TITLE
Updates to the html docs.

### PR DESCRIPTION
**Pull request by Lucarda**
Rewrites of chapter 4 by @porres  and some other fixes (https://github.com/pure-data/pure-data/issues/966).

Here's a [preview](http://lucarda.com.ar/html-preview-2/index.htm) (updated up to ca0f068)

### CODE REVIEW

**Comment by porres** [comment]
Maybe we can use this PR to better revise this htm documentation further. I'm revising the whole thing now... 

I think the chapter needs to be called "External Objects & Libraries" and referenced as such in x1.htm. Then, in the actual page, we have a duplicate of the chapter name, this is inconsistent with the other chapters...

Screen Shot 2019-08-09 at 21 22 56

**Comment by porres** [comment]
it should be something like 

Pd Documentation chapter 4: External Objects & Libraries the go straight to "4.1"

then

**Comment by porres** [comment]
I'm thinking of adding a couple of lines saying what's the chapter is about.

**Comment by porres** [comment]
On second though, keep it all as is, let's just remove "4. External Objects & Libraries". And let the chaoter be called just "externals" indeed!

**Comment by porres** [comment]
In Figure 4.6, can you find a way to just screen shot the information given by loading cyclone and zexy?

**Comment by porres** [comment]
Anyway, I've been editing the section 4 here https://docs.google.com/document/d/1RGY3RkyDPN_WUNekBBPO_4upVt7eNCL6U4F6h111Ecg/edit 

It's already significantly different from what we have here - but hold on please

**Comment by porres** [comment]
I had this issue opened regarding the manual, we could fix it in here as well https://github.com/pure-data/pure-data/issues/705

**Comment by porres** [comment]
@Lucarda , since you're doing such a great job with the new pictures, how about updating the manual pictures with new ones? Most of them are terribly old and outdated way over a decade, such as the testtone.pd screeshot and the preferences windows and everything.

**Comment by porres** [comment]
These are the figures that need to be updated as I see it

- fig1.1.png
- fig8.5.jpg
- fig8.6.jpg
- fig11.1.png
- fig11.2.png
- fig11.3.png
- fig11.4.png

**Comment by porres** [comment]
I had this issue opened regarding the manual, and I've just added a bunch more of them, we could fix them in this PR as well. https://github.com/pure-data/pure-data/issues/705

**Comment by Lucarda** [comment]
> https://docs.google.com/document/d/1RGY3RkyDPN_WUNekBBPO_4upVt7eNCL6U4F6h111Ecg/edit

> It's already significantly different from what we have here - but hold on please

Are you done with the doc?
I can't see it : It asks me for permission.

**Comment by Lucarda** [comment]
@porres here's a preview: http://lucarda.com.ar/html-preview-2/index.htm

It includes the new images and many of the requested changes in #705.

I'm waiting your changes to the "Externals" chapter. (as I said above I cant open the file, it asks me for permission).

**Comment by porres** [comment]
Thanks so much, I'm really happy with all your help :) - can you please remove "4. External Objects & Libraries" from the beggining of section 4 like I said in the first comment/image?

anyway, I've opened the file, here it is... https://docs.google.com/document/d/1RGY3RkyDPN_WUNekBBPO_4upVt7eNCL6U4F6h111Ecg/edit?usp=sharing

**Comment by porres** [comment]
But I'll probably revise it a couple more times before the final release :) - note that I've removed figure numbers, as the manual doesn't actually use them, also removed footnotes. I changed a bit the order of text and figures. Changed the index... so, quite a lot of changes.

**Comment by porres** [comment]
I ended up removing "writing Pd objects in C" and included that text in the beggining of the section - this was basically only a paragraph and a pointer to IOhannes' tutorial. I also added pd-lib-builder as a resource for building externals.

**Comment by porres** [comment]
ping @millerpuckette , I think miller will try to put out a new test version very soon that fixes the sort for macOS, so maybe i can include already the changes we're proposing here.

**Comment by Lucarda** [comment]
> can you please remove "4. External Objects & ...

I didn't change ch4 yet :)

> But I'll probably revise it a couple more times before the final release :)

Is not so fun to manually (and visually) edit an html from a Pdf over and over again so do you think you got a final tomorrow? 

or do you push your edits in html to this branch?

**Comment by porres** [comment]
I'll revise it once again and send you today... I can try and do the next not so fun edits by hand

I thought you had a magic htm editor - I tried them and they just don't work, they really screw everything up

**Comment by Lucarda** [comment]
> I thought you had a magic htm editor - I tried them and they just don't work, they really screw everything up

Yes they do, they also produce horrible code. I've quit WYSIWYG editors a long time ago.

> I'll revise it once again and send you today...

Cool, I then change all the skeleton, internal links and contents. Then if you like you can try edits (if necessary)

**Comment by porres** [comment]
Ok, I finished revising for the day... https://docs.google.com/document/d/1RGY3RkyDPN_WUNekBBPO_4upVt7eNCL6U4F6h111Ecg/edit

I guess we can put it out, include in the next test version, and ask people for further reivision (I'm not a native english speaker)

**Comment by Lucarda** [comment]
> Ok, I finished revising for the day... 

I'll try to have it done in the next 24hs.

:)

**Comment by Lucarda** [comment]
@porres the changes have been made.

:)

preview [here](http://lucarda.com.ar/html-preview-2/index.htm)

**Comment by jmmmp** [comment]
some random suggestions/details:
- should "getting Pd to run" come before "theory of operation"?
- "theory of operation" sounds very dry and scary, how about "Working with Pure Data" or something more inviting?
- use normal case in the titles, instead of small? introduction  -> Introduction 
- some links in http://lucarda.com.ar/html-preview-2/x1.htm are outdated, is it possible to check these automatically?
- also in http://lucarda.com.ar/html-preview-2/x1.htm a reference to downloading gem or getting it with pd-extended, and no mention to deken

**Comment by Lucarda** [comment]
> should "getting Pd to run" come before "theory of operation"?

This is a lot of work changing all the contents of each chapter (x1.htm, x2.htm ... ) and also its internal links. Is not that it can't be done but is rather unnecessary. Also note that at the beginning of chapter 2 it says: 

----------------
- The purpose of this chapter is to describe Pd's design and how it is supposed to work. Practical details about how to obtain, install, and run Pd are described in the [next chapter](url). 
---------------



> some links in http://lucarda.com.ar/html-preview-2/x1.htm are outdated, is it possible to check these automatically?

I don't know if its possible to check automatically. We can correct or erase dead links and also include newer ones.



> also in http://lucarda.com.ar/html-preview-2/x1.htm a reference to downloading gem or getting it with pd-extended, and no mention to deken

This can be corrected.

File: doc/1.manual/x1.htm [hunk 1]
```diff
@@ -96,8 +96,7 @@ <H3> <A id=s2> 1.2. other resources </A> </H3>
 for people interested in graphics, there is a 3D graphics rendering package,
 named GEM, based on OpenGL, written by Mark Danks, adapted to Linux by
 Guenter Geiger, and now maintained by IOhannes zm&ouml;lnig.  You can get
-it from: <A href="http://gem.iem.at/">http://gem.iem.at/</A>, or by
-installing Pd Extended.
+it from: <A href="http://gem.iem.at/">http://gem.iem.at/</A>, or using the externals manager "Deken", see <A href="x4.htm#s2.2"> 4.2.2. How to Download Externals from Pd Vanilla </A>.
```
Comment (umlaeute) [inline doc/1.manual/x1.htm]: i'm not overly fond of installing complex externals with lots of dependencies (like `Gem`) via `deken`.
apart from that, you should remove the leading/trailing spaces in the link.

Comment (Lucarda) [reply doc/1.manual/x1.htm]: > you should remove the leading/trailing spaces in the link.

you mean the trailing spaces here : `> 4.2.2. How to Download Externals from Pd Vanilla </A>.` ?

**Comment by Lucarda** [comment]
It is a typo.
I ment https://github.com/pure-data/pure-data/pull/755

**Pull request merged**